hack/create-identity-secret.sh

```diff
@@ -28,8 +28,17 @@ make --directory="${REPO_ROOT}" "${KUBECTL##*/}"
 # shellcheck source=hack/parse-prow-creds.sh
 source "${REPO_ROOT}/hack/parse-prow-creds.sh"
 
-if ! "${KUBECTL}" get secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" > /dev/null; then
+if ! "${KUBECTL}" get secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" > /dev/null && [[ -n "${AZURE_CLIENT_SECRET:-}" ]]; then
   "${KUBECTL}" create secret generic "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" --from-literal=clientSecret="${AZURE_CLIENT_SECRET}"
 fi
 
-"${KUBECTL}" label secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" "clusterctl.cluster.x-k8s.io/move-hierarchy=true" --overwrite=true
+if "${KUBECTL}" get secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" > /dev/null; then
+  "${KUBECTL}" label secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" "clusterctl.cluster.x-k8s.io/move-hierarchy=true" --overwrite=true
+fi
+
+if ! "${KUBECTL}" get secret "${ASO_CREDENTIAL_SECRET_NAME}" > /dev/null; then
+  "${KUBECTL}" create secret generic "${ASO_CREDENTIAL_SECRET_NAME}" --from-literal=AZURE_SUBSCRIPTION_ID="${AZURE_SUBSCRIPTION_ID}" --from-literal=AZURE_TENANT_ID="${AZURE_TENANT_ID}" --from-literal=AZURE_CLIENT_ID="${AZURE_CLIENT_ID}" --from-literal=AUTH_MODE="${ASO_CREDENTIAL_SECRET_MODE:-workloadidentity}"
```

Comment:
I think using `AZURE_CLIENT_ID` here is what's causing issues in the test where it should be `AZURE_CLIENT_ID_USER_ASSIGNED_IDENTITY`, but that isn't exported here. Since that is usually used directly from a template in an AzureClusterIdentity, I think I'll revert these changes and instead define this Secret in the template in the cluster-autoscaler repo. That Secret won't contain anything we don't already put in an AzureClusterIdentity we define in other templates, so that seems reasonable to me.